trl/rewards/format_rewards.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020-2025 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+
+def think_answer_format_reward(completions: list[list[dict[str, str]]], **kwargs) -> list[float]:
```

Comment:
FYI this particular format is legacy from the [R1 tech report](https://arxiv.org/pdf/2501.12948):

![Screenshot 2025-05-09 at 19 36 38](https://github.com/user-attachments/assets/b055795d-b41e-4052-9208-c6b5f2ace844)

However, if you distill the R1 model itself, you find it only produces `<think>` and `</think>` tags, i.e. no `<answer>` tags at all, e.g. : https://huggingface.co/datasets/open-r1/OpenR1-Math-220k/viewer/default/train?row=0

This I recommend we change this reward function to simply `think_format_reward` and enforce `<think>\nThought section\n</think>\nSolution section.` This has the added advantage of being easier to learn and is consistent with policies initialised from R1 traces

Reply:
Sounds good! done in 335bcc7a1ae86827f69b918ac369bf8878daeea0